packages/embed/src/factories/create-popover/create-popover.ts

```diff
@@ -97,6 +97,48 @@ const buildTooltip = (message: string, hide: () => void) => {
   return container
 }
 
+const LOCAL_STORAGE_NOTIFICATION_DATA = 'tfNotificationData'
+
+type LocalStorageCUIData = {
+  formId: string
+  expireDate: number
+}
+const today = new Date().getTime()
+
+const cuiLocaslStorageDate = typeof window !== 'undefined' && localStorage.getItem(LOCAL_STORAGE_NOTIFICATION_DATA)
+
+const getLocalStorageData = () => cuiLocaslStorageDate && JSON.parse(cuiLocaslStorageDate)
+
+function saveToLocalstorage(data: LocalStorageCUIData) {
+  if (!data) return null
+  typeof window != 'undefined' && localStorage.setItem(LOCAL_STORAGE_NOTIFICATION_DATA, JSON.stringify(data))
+}
+
+function canBuildCUIDot(formId: string, days: number) {
+  const localStorageData = getLocalStorageData()
+  if (localStorageData && localStorageData.expireDate) {
+    return today > localStorageData.expireDate
+  } else {
+    // save data to localstorage
+    const expireDate = new Date()
+    expireDate.setDate(expireDate.getDate() + days)
+
+    saveToLocalstorage({
+      formId,
+      expireDate: expireDate.getTime(),
+    })
+  }
+  return true
```

Comment:
Move this into `else` block please. Or just get rid of the `else` block completely.